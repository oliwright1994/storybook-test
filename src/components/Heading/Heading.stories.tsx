import { Stack } from '@chakra-ui/core'
import { Meta } from '@storybook/react/types-6-0'
import React from 'react'

import Heading from './Heading'

export default { component: Heading, title: 'Components/Heading' } as Meta
export const Standard = (): JSX.Element => (
  <Stack>
    <Heading as="h1">H1 Heading</Heading>
    <Heading as="h2">H2 Heading</Heading>
    <Heading as="h3">H3 Heading</Heading>
    <Heading as="h4">H4 Heading</Heading>
    <Heading as="h5">H5 Heading</Heading>
    <Heading as="h6">H6 Heading</Heading>
  </Stack>
)
