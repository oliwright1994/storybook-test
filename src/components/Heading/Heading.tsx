import { Heading as ChakraHeading } from '@chakra-ui/core'
import React, { FC } from 'react'

interface IHeading {
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
}

const Heading: FC<IHeading> = ({ children, as = 'h1' }) => {
  return <ChakraHeading as={as}>{children}</ChakraHeading>
}

export default Heading
