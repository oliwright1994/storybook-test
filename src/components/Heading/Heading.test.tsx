import { screen } from '@testing-library/react'
import React from 'react'
import { renderWithBrightTheme } from 'testing/helpers/themeHelper'

import Heading from '.'

describe('<Heading />', () => {
  it('should render as expected', () => {
    const { container } = renderWithBrightTheme(
      <>
        <Heading>Heading One</Heading>
        <Heading as="h2">Heading Two</Heading>
      </>
    )
    const headingNodes = screen.getAllByText(/heading/i)

    expect(headingNodes).toHaveLength(2)
    expect(container).toMatchSnapshot()
  })

  it('should render with the correct element tag', () => {
    const { container } = renderWithBrightTheme(
      <>
        <Heading>H1 Heading</Heading>
        <Heading as="h2">H2 Heading</Heading>
        <Heading as="h3">H3 Heading</Heading>
        <Heading as="h4">H4 Heading</Heading>
        <Heading as="h5">H5 Heading</Heading>
        <Heading as="h6">H6 Heading</Heading>
      </>
    )
    for (let i = 1; i < 7; i += 1) {
      const regExp = new RegExp(`H${i}`, 'g')
      expect(screen.getByText(regExp).nodeName).toEqual(`H${i}`)
    }
    expect(container).toMatchSnapshot()
  })
})
