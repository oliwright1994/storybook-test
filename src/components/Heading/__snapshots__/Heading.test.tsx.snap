// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Heading /> should render as expected 1`] = `
<div>
  <h1
    class="css-p5x3r9"
  >
    Heading One
  </h1>
  <h2
    class="css-p5x3r9"
  >
    Heading Two
  </h2>
</div>
`;

exports[`<Heading /> should render with the correct element tag 1`] = `
<div>
  <h1
    class="css-p5x3r9"
  >
    H1 Heading
  </h1>
  <h2
    class="css-p5x3r9"
  >
    H2 Heading
  </h2>
  <h3
    class="css-p5x3r9"
  >
    H3 Heading
  </h3>
  <h4
    class="css-p5x3r9"
  >
    H4 Heading
  </h4>
  <h5
    class="css-p5x3r9"
  >
    H5 Heading
  </h5>
  <h6
    class="css-p5x3r9"
  >
    H6 Heading
  </h6>
</div>
`;
